### PR TITLE
Enabling Ambassador tracing with TracingService, docs and tutorial

### PR DESCRIPTION
**Pull request by alexgervais**
Provides :
- Envoy V1 configuration for enabling tracing on all routes
- Unit and end-to-end tests
- Documentation of `TracingService`, tutorial and Zipkin example

### CODE REVIEW

**Comment by alexgervais** [comment]
Yes, the `request_headers_for_tags` config is something a user might provide. I did not have any use for it myself so I preferred highlighting the empty array in case someone else wants to contribute and make use of it.

**Comment by kflynn** [comment]
@alexgervais Makes sense to me, thanks!

File: ambassador/ambassador/config.py [hunk 1]
```diff
@@ -1535,6 +1546,44 @@ def module_config_ratelimit(self, ratelimit_config):
...
+        host_rewrite = None
+        sources = []
+
+        if tracing_config:
+            for config in tracing_config.values():
+                sources.append(config['_source'])
+                cluster_hosts = config.get("service", None)
+                driver = config.get("driver", None)
+                driver_config = config.get("config", {})
+                host_rewrite = config.get("host_rewrite", None)
```
Comment (containscafeine) [inline ambassador/ambassador/config.py]: looking back at this, should `host_rewrite` be a field in TracingService? And if so, should it be documented?

Comment (alexgervais) [reply ambassador/ambassador/config.py]: If I'm not mistaken it's also an undocumented field of AuthService and RateLimitService. I'm not absolutely sure what was the intent in the original AuthService, but I agree, it should either be documented or removed.

**Pull request merged**